pkg/manifests/manifests.go

```diff
@@ -2140,6 +2140,10 @@ func (f *Factory) MetricsServerDeployment() (*appsv1.Deployment, error) {
 	containers[idx].Args = f.setTLSSecurityConfiguration(podSpec.Containers[0].Args,
 		MetricsServerTLSCipherSuitesFlag, MetricsServerTLSMinTLSVersionFlag)
 
+	// Hash the Kubelet Serving CA Bundle configmap value and propagate it as an annotation to the
+	// deployment's pods to trigger a new rollout when the CA is rotated.
+	dep.Spec.Template.Annotations["monitoring.openshift.io/hash"] = hashConfigMapData(kubeletCABundle)
```

Comment:
I see we also use `monitoring.openshift.io/hash` as a label sometimes https://github.com/search?q=repo%3Aopenshift%2Fcluster-monitoring-operator%20monitoring.openshift.io%2Fhash&type=code, maybe we should unify this?
cc @simonpasquier

Reply:
This annotation has a special meaning IIUC based on which deployment will be rollout https://github.com/openshift/library-go/blob/964bcb3f545c24f15294fd8ba914529bf4fe8c4d/pkg/certs/cert-inspection/certgraphanalysis/collection_options.go#L86

Reply:
Good catch! I'm guilty introducing a different approach using an annotation instead of a label in #2233. Since this change isn't published yet, I'm fine switching to a label instead.

Reply:
Maybe we could still use an annotation after all :) https://github.com/openshift/cluster-monitoring-operator/pull/2242#discussion_r1464544438 wdyt?